hardware/arduino/sam/libraries/HID/HID.h

```diff
@@ -42,16 +42,12 @@
 #define HID_REPORT_DESCRIPTOR_TYPE      0x22
 #define HID_PHYSICAL_DESCRIPTOR_TYPE    0x23
 
-typedef struct __attribute__((packed)) {
-  uint8_t length;
-  const void* descriptor;
-} HID_Descriptor;
-
 class HIDDescriptorListNode {
 public:
   HIDDescriptorListNode *next = NULL;
-  const HID_Descriptor * cb;
-  HIDDescriptorListNode(const HID_Descriptor *ncb) {cb = ncb;}
+  HIDDescriptorListNode(const void *d, uint16_t l) : data(d), length(l) { }
+  uint8_t length;
```

Comment:
> uint16_t?

It should allow for descriptors >256.

> Also this could be const (avr as well)?

Mbah, does it really matter? `l` is passed by copy so the method can't change the "original" parameter in any way.